### PR TITLE
Fix/4235 – Consistent Sorting Of Countries – Alphabetically

### PR DESCRIPTION
**Pull request by tcfos**
## Description
Fixes the inconsistent sorting of country list. 
- on "Andere warnen", countries are sorted as provided by app config (i.e. unsorted)
- on "Ihr Einverständnis", countries are sorted alphabetically

Spec: sort alphabetically.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4235

## Screenshots
IMG_0133
IMG_0132
IMG_0131

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionQRInfoViewModel.swift [hunk 1]
```diff
@@ -8,7 +8,7 @@ import UIKit
 struct ExposureSubmissionQRInfoViewModel {
 
 	init(supportedCountries: [Country]) {
-		countries = supportedCountries
+		countries = supportedCountries.sorted { $0.localizedName.localizedCompare($1.localizedName) == .orderedAscending }
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionQRInfoViewModel.swift:11]: Can we put this in an Extension for [Country] ? 

`extension Array where Element : Country `

Something along those lines.

**Review by marcussc — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionQRInfoViewModel.swift (hunk 1 above)
Comment (tcfos) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionQRInfoViewModel.swift]: Cool idea 👍 
Done with 8058e6e884154897883606e6c2f95fb1b772a5bb

File: src/xcode/ENA/ENA/Source/Models/Exposure/Country.swift [hunk 2]
```diff
@@ -69,3 +69,9 @@ extension Locale {
 	}
 
 }
+
+extension Array where Element == Country {
+	var sortedByLocalizedName: [Country] {
+		return self.sorted { $0.localizedName.localizedCompare($1.localizedName) == .orderedAscending }
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Models/Exposure/Country.swift:75]:
```suggestion
		self.sorted { $0.localizedName.localizedCompare($1.localizedName) == .orderedAscending }
```

Just a hint: you can remove `return` if you want

**Review by nickguendling — APPROVED** [review]

**Review by whiskey — CHANGES_REQUESTED** [review]
Thanks for the fix!

Could you please add a simple unit test? Maybe "Germany, Spain and the Netherlands" ordered in these languages. The ordering should change, based on a quick _guesstimation_ from [this list of translated country names](https://en.wikipedia.org/wiki/List_of_country_names_in_various_languages).

**Review by marcussc — APPROVED** [review]

**Comment by tcfos** [comment]
> Could you please add a simple unit test? 

Added a unit test for sorting countries based on localized country names for all supported languages.

**Review by whiskey — APPROVED** [review]

**Pull request merged**